### PR TITLE
 Adding 2 Points and 3 Points creations method for circle. 

### PR DESCRIPTION
**Pull request by Guruprasad-Rane**
Adding 2 Points and 3 Points creations method for circle.

### CODE REVIEW

File: lcUILua/createActions/circleoperations.lua [hunk 1]
```diff
@@ -1,5 +1,8 @@
 CircleOperations = {}
 CircleOperations.__index = CircleOperations
+local Circle_FirstPoint
+local Circle_SecondPoint
+local Circle_ThirdPoint
```
Comment (feragon) [inline lcUILua/createActions/circleoperations.lua]: You should move those 3 variables in the `_init` function like this:
`self.firstPoint = nil`

Those variables are currently shared between the CircleOperations instances.
You can have conflicts if you have multiple windows open creating circles.

File: lcUILua/createActions/circleoperations.lua [hunk 2]
```diff
@@ -12,19 +15,36 @@ setmetatable(CircleOperations, {
 
 function CircleOperations:_init(id)
     CreateOperations._init(self, id, lc.builder.CircleBuilder, "enterCenter")
-
+    cli_get_text(self.target_widget, true)
```
Comment (feragon) [inline lcUILua/createActions/circleoperations.lua]: This commands prevents the user from entering a coordinate in the command line. 
There isn't a proper way to handle this right now, but you could add a comment explaining that this is the cause of the problem.

**Comment by feragon** [comment]
Thanks for the PR, it's very good!

The name of your variables and functions must begin with a lowercase letter. That helps differentiate them from classes (that begin with an uppercase letter).

File: lcUILua/createActions/circleoperations.lua (hunk 1 above)
Comment (Guruprasad-Rane) [reply lcUILua/createActions/circleoperations.lua]: Ok. I will make the corrections.

**Comment by feragon** [comment]
I think the `_init` method should have another parameter (the options), and if it's defined, we don't ask the user to select an option.
During buttons/menus creation, we have to give the correct option.

**Comment by Guruprasad-Rane** [comment]
Better to start with next command like ARC? or we should first complete coding the toolbar of Circle?

**Comment by feragon** [comment]
It's better to finish Circle first, to keep coherence between the operations

**Comment by Guruprasad-Rane** [comment]
OK. If you can help by setting up the one menu say Create>Circle>2 Point Circle then I can work on the rest.

**Comment by feragon** [comment]
Yes. I'll begin tomorrow

**Comment by feragon** [comment]
@Guruprasad-Rane you can check the circle-menu in my repo.
This breaks the other operations (because I changed the order in CreateOperations)

You can import the commit in your repo with `git pull https://github.com/Feragon/LibreCAD_3.git circle-menu`.

**Comment by Guruprasad-Rane** [comment]
OK. Got it. I can modify other operations one by one.
One more question. How can we issue circle command using dropdown menu Draw>Circle ?

**Comment by feragon** [comment]
You could do it with the `findChild(name)` method. For example:

https://github.com/LibreCAD/LibreCAD_3/blob/master/lcUILua/ui/mainwindow.lua#L99-L103

The name is the one you declared in the .ui file.

**Comment by Guruprasad-Rane** [comment]
Thank you. It worked.
Since rest of the operations are broken, will it be good to work on a branch?

**Comment by feragon** [comment]
Yes, work on a branch. 
If you want to show your progress, you can create a PR too, but it'll not be merged until all operations are fixed.

**Pull request merged**